### PR TITLE
[api-extractor] Add support for TypeScript 3.7

### PR DESCRIPTION
**Pull request by KevinTCoughlin**
Upgrade the compiler engine to `typescript@3.7.2` used by API Extractor (not the RSC compiler used to build API Extractor).

TypeScript 3.7 Release Notes: http://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html

### CODE REVIEW

File: apps/api-extractor/src/api/CompilerState.ts [hunk 1]
```diff
@@ -83,7 +82,7 @@ export class CompilerState {
     const program: ts.Program = ts.createProgram(analysisFilePaths, commandLine.options, compilerHost);
 
     if (commandLine.errors.length > 0) {
-      const errorText: string = TypeScriptMessageFormatter.format(commandLine.errors[0].messageText);
+      const errorText: string = `${commandLine.errors[0].messageText};`;
```
Comment (octogonz) [inline apps/api-extractor/src/api/CompilerState.ts]: We can remove the trailing semicolon.  `formattedErrors.join('; ');` wouldn't have added any delimiter for just one item.

File: apps/api-extractor/src/collector/MessageRouter.ts [hunk 2]
```diff
@@ -169,7 +168,7 @@ export class MessageRouter {
         return;  // ignore noise
     }
 
-    const messageText: string = TypeScriptMessageFormatter.format(diagnostic.messageText);
+    const messageText: string = `${diagnostic.messageText};`;
```
Comment (octogonz) [inline apps/api-extractor/src/collector/MessageRouter.ts]: Same issue with semicolon

Comment (KevinTCoughlin) [reply apps/api-extractor/src/collector/MessageRouter.ts]: Done in https://github.com/microsoft/rushstack/pull/1615/commits/e70461520e16cc4031a9a9e26853f97fde4b88bc.

File: apps/api-extractor/src/api/CompilerState.ts (hunk 1 above)
Comment (KevinTCoughlin) [reply apps/api-extractor/src/api/CompilerState.ts]: Done in https://github.com/microsoft/rushstack/pull/1615/commits/e70461520e16cc4031a9a9e26853f97fde4b88bc.

**Pull request merged**